nixos/modules/services/logging/packetbeat.nix

```diff
@@ -0,0 +1,185 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.packetbeat;
+
+  packetbeatYml = pkgs.writeText "packetbeat.yml" ''
+    name: ${cfg.name}
+    tags: ${builtins.toJSON cfg.tags}
+
+    ${cfg.configFlows}
+    ${cfg.configProtocols}
+    ${cfg.extraConfig}
+  '';
+
+in
+{
+  options = {
+
+    services.packetbeat = {
+
+      enable = mkEnableOption "packetbeat";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.packetbeat;
+        defaultText = "pkgs.packetbeat";
+        example = literalExample "pkgs.packetbeat7";
+        description = ''
+          The packetbeat package to use
+        '';
+      };
+
+      name = mkOption {
+        type = types.str;
+        default = "packetbeat";
+        description = "Name of the beat";
+      };
+
+      tags = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = "Tags to place on the shipped log messages";
+      };
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "packetbeat";
+        description = ''
+          Directory below <literal>/var/lib/</literal> to store packetbeat's
+          own logs and other data. This directory will be created automatically
+          using systemd's StateDirectory mechanism.
+        '';
+      };
+
+      configFlows = mkOption {
+        type = types.lines;
+        default = ''
+          packebeat.flows:
+            timeout: 30s
+            period: 10s
+        '';
+        description = ''
+          Configuration of how packetbeat should handle flows. See
+          <link xlink:href='https://www.elastic.co/guide/en/beats/packetbeat/current/configuration-flows.html'/>
+          for all available configuration options.
+        '';
+      };
+
+      configProtocols = mkOption {
```

Comment:
Would this benefit from a structured type? Maybe `type = with types; attrsOf (either (bool (listOf port)));`.

Reply:
Not impossible, its purely yaml it needs to generate, so a structured type would probably fit. As said in other comments, this was a quick way to getting it usable.